contrib/babelfishpg_common/src/collation.c

```diff
@@ -426,15 +426,39 @@ static void
 init_default_locale(void)
 {
 	if (!default_locale)
-		default_locale = GetConfigOption("babelfishpg_tsql.default_locale", true, false);
+	{
+		const char *val = GetConfigOption("babelfishpg_tsql.default_locale", true, false);
+		if (val)
+		{
+			MemoryContext oldContext = MemoryContextSwitchTo(TopMemoryContext);
+			default_locale = pstrdup(val);
+			MemoryContextSwitchTo(oldContext);
+		}
+	}
+
+	/* babelfishpg_tsql.default_locale should not be changed once babelfish db is initialised. */
+	Assert(strcmp(default_locale, GetConfigOption("babelfishpg_tsql.default_locale", true, false)) == 0);
```

Comment:
is it OK to call strcmp without checking parameters being NULL or not?

Reply:
oh I missed that. Will update the PR!